### DISCUSSION TITLE
Defaulting to Java 21 for new projects

### DISCUSSION THREAD

**Discussion by sdelamo** [micronaut-projects/micronaut-starter#2349] [General]
What do you think if we default to Java 21 to new projects? Not default to 17 would give us more information to [decide whether to drop Java 17 in Micronaut Framework 5](https://github.com/micronaut-projects/micronaut-core/issues/10483). 

ping @micronaut-projects/core-developers

**Comment by dstepanov** [comment]
IMHO we should default to the latest supported version.

**Comment by wetted** [comment]
I have no objection to making the change, and agree with @dstepanov 

I believe it's a simple enough change to make in starter.

**Comment by wetted** [comment]
Completed and merged on #2351
